### PR TITLE
Add shuffle to songs page

### PR DESCRIPTION
**Pull request by murphy52589**
This is my first contribution to open source software so please tell me if something is wrong

**Changes**
A shuffle button has been added under the songs tab in music libraries

**Issues**
https://trello.com/c/ncY5dZrB/26-add-shuffle-all-option-to-music-tab

### CODE REVIEW

File: src/controllers/music/music.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
     <style>
         @media all and (max-width: 32em) {
-
+
```
Comment (thornbill) [inline src/controllers/music/music.html:5]: It looks like there is a lot of whitespace added in this file that shouldn't be there.

File: src/controllers/music/songs.js [hunk 2]
```diff
@@ -8,197 +7,207 @@ import * as userSettings from '../../scripts/settings/userSettings';
 import globalize from '../../scripts/globalize';
 import '../../elements/emby-itemscontainer/emby-itemscontainer';
 import Dashboard from '../../utils/dashboard';
+import {playbackManager} from '../../components/playback/playbackmanager';
 
 /* eslint-disable indent */
```
Comment (thornbill) [inline src/controllers/music/songs.js:12]: It looks like you fixed the indentation issues in this file so these comments can be removed.

**Review by thornbill — CHANGES_REQUESTED** [review]
Congratulations on your first contribution!!! 🎉

I took a quick look and left a couple minor comments. 🙂

File: src/controllers/music/music.html [hunk 3]
```diff
@@ -75,34 +75,36 @@ <h2 class="sectionTitle sectionTitle-cards padded-left">${HeaderFrequentlyPlayed
...
-
+  
         <div is="emby-itemscontainer" id="items" class="itemsContainer padded-left padded-right padded-top vertical-wrap centered"></div>
     </div>
     <div class="pageTabContent" id="songsTab" data-index="5">
         <div class="flex align-items-center justify-content-center flex-wrap-wrap padded-top padded-left padded-right padded-bottom focuscontainer-x">
             <div class="paging"></div>
             <button is="paper-icon-button-light" class="btnSort autoSize" title="${Sort}"><span class="material-icons sort_by_alpha" aria-hidden="true"></span></button>
             <button is="paper-icon-button-light" class="btnFilter autoSize" title="${Filter}"><span class="material-icons filter_list" aria-hidden="true"></span></button>
+            <button is="paper-icon-button-light" class="btnShuffle musicglobalButton" title="${Shuffle}"><span class="material-icons shuffle" aria-hidden="true"></span></button>
```
Comment (dmitrylyzo) [inline src/controllers/music/music.html:97]: Place the shuffle button before `Sort` (as in other views).

File: src/controllers/music/songs.js [hunk 4]
```diff
@@ -8,197 +7,207 @@ import * as userSettings from '../../scripts/settings/userSettings';
...
-                    }
-                    reloadItems(tabContent);
-                }
-
-                function onPreviousPageClick() {
-                    if (isLoading) {
-                        return;
-                    }
+        return savedQueryKey;
+    }
```
Comment (dmitrylyzo) [inline src/controllers/music/songs.js:55]: Are these changes necessary?

File: src/controllers/music/songs.js [hunk 5]
```diff
@@ -8,197 +7,207 @@ import * as userSettings from '../../scripts/settings/userSettings';
...
+        });
+    };
+
+    self.getCurrentViewStyle = function () {
+        return getPageData(tabContent).view;
+    };
+
+    function shuffle() {
+        ApiClient.getItem(ApiClient.getCurrentUserId(), params.topParentId).then(function (item) {
+            getQuery();
```
Comment (dmitrylyzo) [inline src/controllers/music/songs.js:156]: Is `getQuery` necessary?
We may need to use a filter on shuffle at some point, but the latter doesn't support filtering yet.

File: src/controllers/music/songs.js (hunk 4 above)
Comment (murphy52589) [reply src/controllers/music/songs.js:53]: I reverted it. When I started out, something around this threw an error but now it doesn't. Don't know what happened

File: src/controllers/music/music.html (hunk 3 above)
Comment (murphy52589) [reply src/controllers/music/music.html]: Got it

File: src/controllers/music/music.html (hunk 1 above)
Comment (murphy52589) [reply src/controllers/music/music.html]: Not sure what caused that, it's been removed. Maybe Webstorm was feeling saucy

File: src/controllers/music/songs.js [hunk 6]
```diff
@@ -8,197 +8,201 @@ import * as userSettings from '../../scripts/settings/userSettings';
...
+                    id: 'Runtime,AlbumArtist,Album,SortName'
+                }],
+                callback: function () {
+                    getQuery(tabContent).StartIndex = 0;
+                    reloadItems(tabContent);
+                },
+                query: getQuery(tabContent),
+                button: e.target
+            });
+            tabContent.querySelector('.btnShuffle').addEventListener('click', shuffle);
```
Comment (dmitrylyzo) [inline src/controllers/music/songs.js:199]: Move this out of block - that's why it doesn't work until you pressed the sort button

File: src/controllers/music/music.html [hunk 7]
```diff
@@ -2,7 +2,6 @@
 
     <style>
         @media all and (max-width: 32em) {
-
```
Comment (dmitrylyzo) [inline src/controllers/music/music.html:5]: I would leave these blank lines to minimize unnecessary changes.

File: src/controllers/music/music.html [hunk 8]
```diff
@@ -92,10 +78,10 @@ <h2 class="sectionTitle sectionTitle-cards padded-left">${HeaderFrequentlyPlayed
     <div class="pageTabContent" id="songsTab" data-index="5">
         <div class="flex align-items-center justify-content-center flex-wrap-wrap padded-top padded-left padded-right padded-bottom focuscontainer-x">
             <div class="paging"></div>
+            <button is="paper-icon-button-light" class="btnShuffle musicglobalButton" title="${Shuffle}"><span class="material-icons shuffle" aria-hidden="true"></span></button>
```
Comment (dmitrylyzo) [inline src/controllers/music/music.html:81]:
```suggestion
            <button is="paper-icon-button-light" class="btnShuffle autoSize" title="${Shuffle}"><span class="material-icons shuffle" aria-hidden="true"></span></button>
```
In fact, I haven't found `autoSize` style. 🤷‍♂️ 

We also need to hide the button if there is no items.
Like here: https://github.com/jellyfin/jellyfin-web/pull/3627

**Review by dmitrylyzo — COMMENTED** [review]
I don't know where the other commits came from. Could you rebase the branch to jellyfin-web master instead of merging?

**Comment by dmitrylyzo** [comment]
@murphy52589 You need an interactive rebase - when rebasing, remove translation and merge commits to cleanup the history.

When the PR is complete, it would be a good idea to squash the commits.

**Review by dmitrylyzo — APPROVED** [review]
@murphy52589 I have rebased and cleaned up the commits. If you need to continue, please re-checkout branch.
CodeQL is irrelevant.

**Review by thornbill — APPROVED** [review]
